### ISSUE TITLE
Exception when attempting to send an email, Reset Password

### ISSUE THREAD

**Issue by vsc55** [tidusjar/Ombi#3491]
**Describe the bug**
Password reset emails are not sent

**To Reproduce**
1. Go to page login
2. Click on button restart password (I don't know exactly how it appears in the English version)
3. Enter email user
4. Click on button Restore Password
5. No error is displayed on the screen but neither is an email received.

**Expected behavior**
Receive email with link to reset password.

**Logs (Logs directory where Ombi is located)**
https://github.com/tidusjar/Ombi/blob/fc108fcba4a55415c541f308bbb3eba84665aa92/src/Ombi.Notifications/GenericEmailProvider.cs#L47
```
Excepción producida: 'System.InvalidOperationException' en System.Private.Uri.dll
Ombi.Notifications.GenericEmailProvider:Error: Exception when attempting to send an email

System.InvalidOperationException: This operation is not supported for a relative URI.
   at System.Uri.get_DnsSafeHost()
   at System.Uri.get_IdnHost()
   at Ombi.Notifications.GenericEmailProvider.SendAdHoc(NotificationMessage model, EmailNotificationSettings settings) in D:\Source\GitHub\Ombi\src\Ombi.Notifications\GenericEmailProvider.cs:line 47
Excepción producida: 'System.InvalidOperationException' en Ombi.Notifications.dll
Excepción producida: 'System.InvalidOperationException' en System.Private.CoreLib.dll
Excepción producida: 'System.InvalidOperationException' en System.Private.CoreLib.dll
Excepción producida: 'System.InvalidOperationException' en System.Private.CoreLib.dll
Ombi.ErrorHandlingMiddleware:Error: Something bad happened, ErrorMiddleware caught this

System.InvalidOperationException: This operation is not supported for a relative URI.
   at System.Uri.get_DnsSafeHost()
   at System.Uri.get_IdnHost()
   at Ombi.Notifications.GenericEmailProvider.SendAdHoc(NotificationMessage model, EmailNotificationSettings settings) in D:\Source\GitHub\Ombi\src\Ombi.Notifications\GenericEmailProvider.cs:line 47
   at Ombi.Controllers.IdentityController.SubmitResetPassword(SubmitPasswordReset email) in D:\Source\GitHub\Ombi\src\Ombi\Controllers\IdentityController.cs:line 784
   at lambda_method(Closure , Object )
   at Microsoft.Extensions.Internal.ObjectMethodExecutorAwaitable.Awaiter.GetResult()
   at Microsoft.AspNetCore.Mvc.Internal.ActionMethodExecutor.AwaitableObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeNextActionFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.InvokeInnerFilterAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeNextResourceFilter()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Rethrow(ResourceExecutedContext context)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeFilterPipelineAsync()
   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.InvokeAsync()
   at Microsoft.AspNetCore.Builder.RouterMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.StaticFiles.StaticFileMiddleware.Invoke(HttpContext context)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIIndexMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext)
   at Microsoft.AspNetCore.Cors.Infrastructure.CorsMiddleware.InvokeCore(HttpContext context)
   at Ombi.ApiKeyMiddlewear.Invoke(HttpContext context) in D:\Source\GitHub\Ombi\src\Ombi\Middleware\ApiKeyMiddlewear.cs:line 51
   at Ombi.ErrorHandlingMiddleware.Invoke(HttpContext context) in D:\Source\GitHub\Ombi\src\Ombi\Middleware\ErrorHandlingMiddlewear.cs:line 24
```

**Desktop (please complete the following information):**
 - OS: Linux and Windows.

**Ombi Version (please complete the following information):**
 - Version 3.0.5115 

**Additional context**
This is about a bug reported by juanjo900 on Discord.
This exception only occurs when sending the email to reset the users password. As commented by juanjo900 the emails of notices of new added movies and requests arrive correctly.

The smtp configuration is:
{
  "DisableCertificateChecking": false,
  "DisableTLS": false,
  "Host": "smtp.gmail.com",
  "Port": 465
 }

**Comment by vsc55** [comment]
The error occurs when ApplicationUrl is not defined.

**Comment by vsc55** [comment]
The exception is generated in:
https://github.com/tidusjar/Ombi/blob/fc108fcba4a55415c541f308bbb3eba84665aa92/src/Ombi.Notifications/GenericEmailProvider.cs#L47

The problem is in `url.IdnHost` since if the URL is not defined or there is a URL that is not valid, for example 'url=/token?token=' the exception is generated.

**Comment by vsc55** [comment]
See what you think about the solution:
https://github.com/vsc55/Ombi/commit/13c6788ad703a960f0789ff188a2f2310bb3efb7

With this the issues are solved: #3491, #2865

**Comment by tidusjar** [comment]
Just made some comments on the changes
